morf-core/src/main/java/org/alfasoftware/morf/upgrade/UpgradePath.java

```diff
@@ -367,12 +374,27 @@ public UpgradePath create(List<UpgradeStep> steps, ConnectionResources connectio
 
 
     @Override
-    public UpgradePath create(List<UpgradeStep> steps,  ConnectionResources connectionResources, GraphBasedUpgradeBuilder graphBasedUpgradeBuilder) {
+    public UpgradePath create(List<UpgradeStep> steps,
+                              ConnectionResources connectionResources,
+                              GraphBasedUpgradeBuilder graphBasedUpgradeBuilder,
+                              List<String> optimisticLockingInitialisationSql) {
       UpgradeStatusTableService upgradeStatusTableService = upgradeStatusTableServiceFactory.create(connectionResources);
       return new UpgradePath(upgradeScriptAdditions, steps, connectionResources,
-              upgradeStatusTableService.updateTableScript(UpgradeStatus.NONE, UpgradeStatus.IN_PROGRESS),
+              combineInitialisationSql(upgradeStatusTableService, optimisticLockingInitialisationSql),
               upgradeStatusTableService.updateTableScript(UpgradeStatus.IN_PROGRESS, UpgradeStatus.COMPLETED),
               graphBasedUpgradeBuilder);
     }
+
+
+    /**
+     * Creates the initialisation SQL by combining the creation of the upgrade status table, followed by the optimised locking SQL.
+     * The optimised locking statement should always come after, as the locking may be dependent on the upgrade status table.
+     */
+    private List<String> combineInitialisationSql(UpgradeStatusTableService upgradeStatusTableService,
+                                                  List<String> optimisticLockingInitialisationSql) {
+      List<String> updateTableSql = upgradeStatusTableService.updateTableScript(UpgradeStatus.NONE, UpgradeStatus.IN_PROGRESS);
+      updateTableSql.addAll(optimisticLockingInitialisationSql);
+      return updateTableSql;

```

Comment:
This relies on `upgradeStatusTableService.updateTableScript()` returning a mutable List, which you can add to.
First, `upgradeStatusTableService.updateTableScript()` shuld return an immutable list, second, this needs to build its own (immutable) list.